### PR TITLE
Build process enhancements

### LINKED ISSUE #24: Large bundle size
This library comes at a whopping 113.1kB minified https://bundlephobia.com/result?p=@typeform/embed@0.16.0 

That is not very good for just showing a form on a website.

### LINKED ISSUE #40: [BUG] Zone.js has detected that ZoneAwarePromise `(window|global).Promise` has been overwritten
On iOS platform the cordova application don't start and in console output I see next error:

Error: Zone.js has detected that ZoneAwarePromise `(window|global).Promise` has been overwritten.
Most likely cause is that a Promise polyfill has been loaded after Zone.js (Polyfilling Promise api is not necessary when zone.js is loaded. If you must load one, do so before loading zone.js.)

Reproduced on iOS app start.

### PR DESCRIPTION
**Pull request by prilutskiy**
Resolves #40 
Resolves #24

### CODE REVIEW

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/8e6c2399851beffb0af8abc385cba4d2cc371011/embed.js

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/471dbe9f9af247fd4ef7b4dabe80f814b394bb4f/embed.js

File: README.md [hunk 1]
```diff
@@ -14,18 +14,29 @@ Requirements:
...
 
 ```bash
 yarn add @typeform/embed
 ```
+The NPM version doesn't include neither React, ReacDOM nor polyfills. It's a responsibility of the application to provide those deps as peer dependencies and control the environment polyfilling process.
+
+#### CDN script
 
-If you're not, you can still use it with our [CDN](https://embed.typeform.com/embed.js):
+You can also consume the library fromt our [CDN](https://embed.typeform.com/embed.js):
```
Comment (prilutskiy) [inline README.md:29]: typo

File: .gitignore [hunk 2]
```diff
@@ -51,6 +51,7 @@ coverage
 
 dist
 lib
```
Comment (prilutskiy) [inline .gitignore:53]: dist & lib are old ones and need to be removed

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/611c6d4f10cd956c8eedde3247bbc26b022c3d79/embed.js

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/611c6d4f10cd956c8eedde3247bbc26b022c3d79/embed.js

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/ccb1bcd96c02a12e298e95f931433e4583dadd94/embed.js

File: .huskyrc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "hooks": {
+    "pre-commit": "yarn run lint",
+    "pre-push": "yarn test:unit",
```
Comment (prilutskiy) [inline .huskyrc:4]: cc @mathio

File: cypress.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "chromeWebSecurity": false,
   "integrationFolder": "./e2e/spec",
   "screenshotsFolder": "./e2e/screenshots",
-  "baseUrl": "http://localhost:8080/",
+  "baseUrl": "http://localhost:5000/",
```
Comment (mathio) [inline cypress.json:5]: Are you running demo at `5000` now? I dont see that change.

File: README.md [hunk 5]
```diff
@@ -14,18 +14,29 @@ Requirements:
...
+#### CDN script
 
-If you're not, you can still use it with our [CDN](https://embed.typeform.com/embed.js):
+You can also consume the library from our [CDN](https://embed.typeform.com/embed.js):
 
 ```html
 <script src="https://embed.typeform.com/embed.js"></script>
 <!-- this script exposes the API to `window.typeformEmbed` -->
 ```
+The CDN version has everything bundled, so the script is ready to use out-of the box. It contains react, react-dom, and polyfills. If you believe everything is already injected in your environment globals you can use a more lightweight version, ***although it's not recommended***:
```
Comment (lfons) [inline README.md:35]: Just to reiterate a bit on what we already commented on other occasions. I believe that we should be able to get rid of React and React DOM. It seems that the functionality provided by the embed SDK could be implemented without these third-party libraries. Not sure if we already have a ticket for this, or if you think we should create one. WDYT @prilutskiy @mathio ?

File: src/core/views/components/spinner.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import Spin from 'spin.js'
+import { Spinner as Spin } from 'spin.js'
```
Comment (lfons) [inline src/core/views/components/spinner.js:3]: I noticed that the spinner was not spinning for me. Is this also happening for you guys?

Comment (mathio) [reply src/core/views/components/spinner.js:3]: Yes, I see the same issue. I think that should be fixed.

File: README.md (hunk 5 above)
Comment (mathio) [reply README.md:35]: I agree however that would require rewrite of the library. We do not just use JSX but also life cycle methods. It is out of scope for now but definitely something we should revisit later.

File: cypress.json (hunk 4 above)
Comment (prilutskiy) [reply cypress.json:5]: "start": "yarn serve -S demo"

[serve](https://www.npmjs.com/package/serve) uses 5000 by default

File: README.md (hunk 5 above)
Comment (prilutskiy) [reply README.md:35]: Absolutely agree with you guys, that's a great point! Let's brainstorm this and analyze the cost/benefits.

File: src/core/views/components/spinner.js (hunk 6 above)
Comment (prilutskiy) [reply src/core/views/components/spinner.js:3]: On it

**Comment by travis-tf** [comment]
*[BOT]* Preview available: https://embed.typeform.com/9037905c620469f75bf8fba66215cf58ff87e35f/embed.js

**Pull request merged**